### DISCUSSION TITLE
No `address_place_id` of countries in `place_addressline`

### DISCUSSION THREAD

**Discussion by foodev** [osm-search/Nominatim#2448] [Q&A] [answered]
As described in https://github.com/osm-search/Nominatim/issues/2439 a few days ago, I'm trying to get a list of all the places (`boundary=administrative`) with their immediate children (and the children's children etc.).

@lonvia has already [helped](https://github.com/osm-search/Nominatim/issues/2439#issuecomment-912627848) me a lot with a PostgreSQL query:

```sql
SELECT p2.name -> 'name'
FROM placex p1, placex p2, place_addressline pa
WHERE p1.osm_id = 62422
AND p1.osm_type = 'R'
AND pa.address_place_id = p1.place_id
AND pa.place_id = p2.place_id
AND pa.isaddress
AND pa.fromarea
AND p2.admin_level = 9
```

This works fine except for the country level. For example, if I use `osm_id=51477` (Germany) and `admin_level=4`, nothing is found. I would expect to get a list of the federal states of Germany.

Apparently records with `address_place_id=<the_place_id_of_the_country>` are missing for all countries. I wonder if this is intentional or configurable. I tried to edit [address-levels.json](https://github.com/osm-search/Nominatim/blob/master/settings/address-levels.json) but without success.

**Comment by lonvia** [comment, marked as answer]
Nominatim stops the address hierarchy computation at admin_level=2 as a matter of optimisation. You need to use `country_code` to bootstrap the process at admin_level=3 (or whatever the first level is after the country).

**Reply by foodev** [reply]
Thanks, this works great.
